File: web/src/lib/wallet-movements.ts
export type WalletTx = {
  id?: string;
  type?: string;
  amountCdf?: number;
  description?: string;
  reference?: string;
  createdAt?: string;
};

export function isWalletRecharge(tx: WalletTx): boolean {
  const desc = tx.description ?? "";
  const ref = tx.reference ?? "";
  if (desc.startsWith("Recharge")) return true;
  if (ref.startsWith("topup_")) return true;
  if (desc.includes("Annulation retrait")) return true;
  return false;
}

export function isWalletWithdraw(tx: WalletTx): boolean {
  const desc = tx.description ?? "";
  const ref = tx.reference ?? "";
  if (desc.startsWith("Retrait")) return true;
  if (ref.startsWith("withdraw_")) return true;
  return false;
}

export function isMobileMoneyMovement(tx: WalletTx): boolean {
  return isWalletRecharge(tx) || isWalletWithdraw(tx);
}

export function walletTxLabel(tx: WalletTx): string {
  if (tx.description?.trim()) return tx.description;
  if (isWalletRecharge(tx)) return "Recharge";
  if (isWalletWithdraw(tx)) return "Retrait";
  return tx.type === "CREDIT" ? "Crédit" : "Débit";
}

File: mobile/lib/core/wallet/wallet_movements.dart
/// Classification des mouvements portefeuille (recharge / retrait Mobile Money).
class WalletMovements {
  WalletMovements._();

  static bool isRecharge(Map<String, dynamic> tx) {
    final desc = tx['description']?.toString() ?? '';
    final ref = tx['reference']?.toString() ?? '';
    if (desc.startsWith('Recharge')) return true;
    if (ref.startsWith('topup_')) return true;
    if (desc.contains('Annulation retrait')) return true;
    return false;
  }

  static bool isWithdraw(Map<String, dynamic> tx) {
    final desc = tx['description']?.toString() ?? '';
    final ref = tx['reference']?.toString() ?? '';
    if (desc.startsWith('Retrait')) return true;
    if (ref.startsWith('withdraw_')) return true;
    return false;
  }

  static bool isMobileMoneyMovement(Map<String, dynamic> tx) =>
      isRecharge(tx) || isWithdraw(tx);

  static String label(Map<String, dynamic> tx) {
    final desc = tx['description']?.toString();
    if (desc != null && desc.isNotEmpty) return desc;
    if (isRecharge(tx)) return 'Recharge';
    if (isWithdraw(tx)) return 'Retrait';
    final type = tx['type']?.toString() ?? '';
    return type == 'CREDIT' ? 'Crédit' : 'Débit';
  }
}
